articles/10-tips-for-working-with-node.js/index.md

```diff
@@ -0,0 +1,295 @@
+
+To use HTTP and client server in Node.js you must use `require ('http');`. Let's look at a simple Hello World HTTP Server module that will listen on port number 3000.
+```js
+var http = require('http')
+
+var server = http.createServer((function(request, response) {
+
+    response.writeHead(200, { "Content-Type\": "text/plain" });
+
+    response.end("Hello World\n");
+
+}));
+server.listen(3000);
+```
+When you type localhost:3000 on your browser, The output will be "Hello
+World".
+
+Check out this [article](https://www.section.io/engineering-education/http-requests-nodejs/) to find out more about on working with HTTP and Node.js.
+
+## 4. Embracing Asynchronous Node.js Functions
+
+Asynchronous is heavily used in Node.js to ensure non-blocking operations flow. It enables a system to handle thousands of concurrent requests and process gigabytes of data with a small amount of RAM. 
+
+Asynchronous I/O permits other processing to continue even before the first transmission has finished. Asynchronous helps to avoid the so-called "Call-back Hell"
+
+Example of async/await in Node.js:
+```js
+async function myFunction(inputValue) {
+
+    try {
+
+        const a = await asyncFunc1('value');
+
+        const b = await asyncFunc2(a);
+
+        const c = syncFunc3(b);
+
+        return await asyncFunc4(c);
+
+    } catch (ex) {
+
+        *// handle exception*
+
+    }
+
+}
+```
+In this scenario, we can see that functions 1, 2 and 4 are
+asynchronous.
+
+## 5. Handling Node.js Errors
+
+The common Node.js errors are:
+
+![Node.js_error_types](/engineering-education/10-tips-for-working-with-node.js/node.js_error_types.png)
+
+In Node.js, errors are handled through exceptions. For example, if we
+have an error that would occur when we divide a number by zero, our
+Node.js application will crash, so then we need to handle the error to
+continue with the normal execution of our application i.e.
+```js
+try {
+
+    var m = 1;
+
+    var n = 1 / 0;
+
+} catch (err) {
+
+    *//Handle the error here.*
+
+    *//the application does do something to execute the error without having to make our application crash*
+
+}
+```
+
+## 6. Debugging Node.js Programs
+
+As a developer, you need to constantly check bugs in your apps and scripts.
+Debugging helps in tracing coding error back to its source making it easier to find bugs efficiently.The commonly used approach for debugging Node.js application is `console.log`.
+```js
+*// writing a hello world to console: *
+console.log(\"hello world\");
+```
+`Console.log` is provided by internet browsers such as Chrome DevTools, it has debug protocols ported into a Node.js module and can be used to debug Node apps. 
+
+It requires you to `install node -inspector` (`npm install -g node-inspector`) then run ` node-debug app.js --debug-brk`. 
+This will open up Chrome Developer Tools and you can start debugging your app. 
+
+Alternatively, you can use Node.js' built-in debugger by running `node debug app.js`.
```

Comment:
See console message: `node debug` is deprecated. Please use `node inspect` instead. Just change the word and it works exactly the same.